operator-framework/src/test/java/io/javaoperatorsdk/operator/PrimaryToSecondaryDependentIT.java

```diff
@@ -0,0 +1,76 @@
+package io.javaoperatorsdk.operator;
+
+import java.time.Duration;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension;
+import io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentCustomResource;
+import io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentReconciler;
+import io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentSpec;
+
+import static io.javaoperatorsdk.operator.sample.primarytosecondaydependent.ConfigMapReconcilePrecondition.DO_NOT_RECONCILE;
+import static io.javaoperatorsdk.operator.sample.primarytosecondaydependent.PrimaryToSecondaryDependentReconciler.DATA_KEY;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+class PrimaryToSecondaryDependentIT {
+
+  public static final String TEST_CONFIG_MAP_NAME = "testconfigmap";
+  public static final String TEST_CR_NAME = "test1";
+  public static final String TEST_DATA = "testData";
+  public
+
+  @RegisterExtension LocallyRunOperatorExtension operator =
+      LocallyRunOperatorExtension.builder()
+          .withReconciler(new PrimaryToSecondaryDependentReconciler())
+          .build();
+
+  @Test
+  void testPrimaryToSecondaryInDependentResources() {
+    var reconciler = operator.getReconcilerOfType(PrimaryToSecondaryDependentReconciler.class);
+    var cm = operator.create(configMap(DO_NOT_RECONCILE));
+    operator.create(testCustomResource());
+
+    await().pollDelay(Duration.ofMillis(250)).untilAsserted(() -> {
+      assertThat(reconciler.getNumberOfExecutions()).isPositive();
+      assertThat(operator.get(Secret.class, TEST_CR_NAME)).isNull();
+    });
+
+    cm.setData(Map.of(DATA_KEY, TEST_DATA));
+    operator.replace(cm);
+    var executions = reconciler.getNumberOfExecutions();
```

Comment:
It would make more sense to record the number of executions before updating the CR, though in practice it probably doesn't matter much.

Reply:
ahh right, changed.